/*
 * Copyright 2021 Google LLC.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package yggdrasil_decision_forests.model.distributed_decision_tree.proto;

import "yggdrasil_decision_forests/learner/decision_tree/decision_tree.proto";
import "yggdrasil_decision_forests/model/decision_tree/decision_tree.proto";

// Definition and statistics of a given split.
message Split {
  optional decision_tree.proto.NodeCondition condition = 1;
  optional decision_tree.proto.LabelStatistics label_statistics_neg = 2;
  optional decision_tree.proto.LabelStatistics label_statistics_pos = 3;
}

// List of splits. One for each currently open node.
message SplitPerOpenNode {
  repeated Split splits = 1;
}
